pilot/pkg/proxy/envoy/v1/watcher.go

```diff
@@ -263,6 +263,7 @@ func (proxy envoy) args(fname string, epoch int) []string {
 		"--service-cluster", proxy.config.ServiceCluster,
 		"--service-node", proxy.node,
 		"--max-obj-name-len", fmt.Sprint(MaxClusterNameLength), // TODO: use MeshConfig.StatNameLength instead
+		"--concurrency", fmt.Sprint(proxy.config.Concurrency),
```

Comment:
I'm a bit worried about this - envoy docs says ' If not specified defaults to the number of hardware threads on the machine.'

I'm afraid the change will break more than it fixes - instead of a reasonable default based on each node, it'll have the same config everywhere (if set in mesh config).

Can you leave the default value to 0, and only set the flag if config is explicitly set ?

Reply:
@costinm Yes, I think this can cause an issue if specified in the default mesh config. I can set the concurrency value to 0 by default and have a check inside of envoy args (here: https://github.com/istio/istio/blob/master/pilot/pkg/proxy/envoy/v1/watcher.go#L267) to add the concurrency flag if it is not 0. This will ensure that the flag is only added if it's specified by the user. What do you think of this approach?